JOSESwift/Sources/Decrypter.swift

```diff
@@ -75,11 +75,11 @@ public struct Decrypter {
     ///   - contentDecryptionAlgorithm: The algorithm used to decrypt the JWE's payload.
     /// - Returns: A fully initialized `Decrypter` or `nil` if provided key is of the wrong type.
     public init?<KeyType>(keyDecryptionAlgorithm: AsymmetricKeyAlgorithm, keyDecryptionKey kdk: KeyType, contentDecryptionAlgorithm: SymmetricKeyAlgorithm) {
+        guard type(of: kdk) is RSADecrypter.KeyType.Type else {
+            return nil
+        }
         switch (keyDecryptionAlgorithm, contentDecryptionAlgorithm) {
-        case (.RSA1_5, .A256CBCHS512):
-            guard type(of: kdk) is RSADecrypter.KeyType.Type else {
```

Comment:
As there is a PR for direct encryption (see #92) I think this should remain here since the `keyDecryptionKey` can be of type type Data as well. (Just to avoid that you need to resolve any merge conflicts.)

Reply:
ok

Reply:
ok

Reply:
Fixed